lnbits/core/views/wallet_api.py

```diff
@@ -62,6 +76,39 @@ async def api_wallets_paginated(
     return page
 
 
+@wallet_router.put("/share/invite")
+async def api_invite_wallet_share(
+    data: WalletSharePermission, key_info: WalletTypeInfo = Depends(require_admin_key)
+) -> WalletSharePermission:
+    if data.wallet_id != key_info.wallet.id:
+        raise HTTPException(
+            HTTPStatus.FORBIDDEN, "Wallet ID does not match the admin key's wallet."
+        )
+    return await invite_to_wallet(key_info.wallet, data)
+
+
+@wallet_router.delete("/share/invite/{share_request_id}")
+async def api_reject_wallet_invitation(
+    share_request_id: str, key_info: WalletTypeInfo = Depends(require_admin_key)
+) -> SimpleStatus:
+    await reject_wallet_invitation(key_info.wallet.user, share_request_id)
+    return SimpleStatus(success=True, message="Invitation rejected.")
+
+
+@wallet_router.put("/share")
+async def api_accept_wallet_share_request(
+    data: WalletSharePermission, key_info: WalletTypeInfo = Depends(require_admin_key)
+) -> WalletSharePermission:
+    return await update_wallet_share_permissions(key_info.wallet, data)
```

Comment:
i think you should handle the ValueError here and return HttpExceptions. also did you check if the key_info.wallet == data.wallet